bibliopixel/animation/base.py

```diff
@@ -56,21 +34,29 @@ def stopThread(self, wait=False):
                 self._thread.join()
 
     def __enter__(self):
+        # DEPRECATED in favor of self.run_cleanup()
         return self
 
     def __exit__(self, type, value, traceback):
+        # DEPRECATED in favor of self.run_cleanup()
+        self.cleanup()
+
+    def cleanup(self):
         self.stopThread(wait=True)
         self._led.all_off()
         self._led.push_to_driver()
         self._led.waitForUpdate()
 
-    def cleanup(self):
-        return self.__exit__(None, None, None)
+    def run_cleanup(self, **kwds):
```

Comment:
It's now gone in the later changes I pushed - because `run` cleaned up anyway, except when there was an exception, and I fixed that second part.